game-core/src/test/java/games/strategy/triplea/delegate/battle/FakeBattleState.java

```diff
@@ -82,6 +82,18 @@
     return dependentUnits;
   }
 
+  @Override
+  public Collection<Unit> getUnits(final EnumSet<Side> sides) {
+    final Collection<Unit> units = new ArrayList<>();
```

Comment:
Seeing this same code 3 times is raising real questions :grin: 

Why does FakeBattleState need to have this implementation? Can we avoid it and perhaps use a true mock? This is kinda concerning code as our test code has implementation that arguably should have a test of its own.

Are there options where we can remove this method and/or class?

Reply:
I created FakeBattleState so that I didn't have to mock as many things.  Instead of having to specify all the required states, I can just use this real object that has default values and then override those defaults as needed.

I can't remove the method because it is part of the `BattleState` interface and this implements that interface.

As for code in 3 places, it is looking more and more like I'll be deleting `BattleSteps`.  There's 2-5 more steps that need to be converted but then the code for `BattleSteps` will basically be a list of objects that are sorted.  At that point, I'm thinking I just need a helper class to build the list of objects and then `MustFightBattle` can grab those objects, sort them, and then map them to their step names.  That will remove one of the `BattleState` implementations.

Reply:
I'll certainly say that mocks are a testing smell and need to be used strategically. Mocking input can be good (though still it's better to use value objects for input),  injecting mock implementations is generally bad.

Fakes are certainly a class of test object of their own, similar to stubs, but often one would want a very simplified implementation and not a real implementation. I think we probably would want to get rid of `FakeBattleState` in the medium term. I think we can see how this evolves knowing that there are still some smells lingering around.